From 74826845e37870ef665b2440705330901bdde4aa Mon Sep 17 00:00:00 2001
From: yingzhex <yingzhenx.li@intel.com>
Date: Tue, 7 Jan 2020 12:07:27 +0800
Subject: [PATCH] Workaround the audio and video flickering issue during
 seeking

Pause the video playback during seeking.

Tracked-On: OAM-88796
Signed-off-by: yingzhex <yingzhenx.li@intel.com>
---
 src/com/android/gallery3d/app/CommonControllerOverlay.java | 7 +++++++
 src/com/android/gallery3d/app/MoviePlayer.java             | 6 ++++++
 2 files changed, 13 insertions(+)

diff --git a/src/com/android/gallery3d/app/CommonControllerOverlay.java b/src/com/android/gallery3d/app/CommonControllerOverlay.java
index 9adb4e7..97085b9 100644
--- a/src/com/android/gallery3d/app/CommonControllerOverlay.java
+++ b/src/com/android/gallery3d/app/CommonControllerOverlay.java
@@ -132,6 +132,13 @@ public abstract class CommonControllerOverlay extends FrameLayout implements
         return view;
     }
 
+    public boolean getPlayingState() {
+        if(mState == State.PLAYING){
+            return true;
+        }
+        return false;
+    }
+
     @Override
     public void setListener(Listener listener) {
         this.mListener = listener;
diff --git a/src/com/android/gallery3d/app/MoviePlayer.java b/src/com/android/gallery3d/app/MoviePlayer.java
index f6bd367..a3c478d 100644
--- a/src/com/android/gallery3d/app/MoviePlayer.java
+++ b/src/com/android/gallery3d/app/MoviePlayer.java
@@ -378,6 +378,9 @@ public class MoviePlayer implements
     @Override
     public void onSeekStart() {
         mDragging = true;
+        if(mController.getPlayingState()){
+            mVideoView.pause();
+        }
     }
 
     @Override
@@ -388,6 +391,9 @@ public class MoviePlayer implements
     @Override
     public void onSeekEnd(int time, int start, int end) {
         mDragging = false;
+        if(mController.getPlayingState()){
+            mVideoView.start();
+        }
         mVideoView.seekTo(time);
         setProgress();
     }
-- 
2.7.4

